lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/util/batch/AzureBatchHelper.java

```diff
@@ -169,4 +190,39 @@ public void submitTask(final String jobId, final String taskId, final URI jobJar
   public String getAzureBatchJobId() {
     return System.getenv(AZ_BATCH_JOB_ID_ENV);
   }
+
+  private TaskContainerSettings createTaskContainerSettings(final String dockerContainerId) {
+    if (!this.areContainersEnabled) {
+      return null;
+    }
+
+    StringBuilder portMappings = new StringBuilder();
+    Iterator<Integer> iterator = this.ports.iterator();
+    while (iterator.hasNext()) {
+      Integer port = iterator.next();
+      portMappings.append(String.format("-p %d:%d ", port, port));
+    }
```

Comment:
maybe just
```java
final StringBuilder runOptions = new StringBuilder(String.format(
    "-d --rm --name %s --env %s=%s ",
    dockerContainerId,
    ContainerBasedLocalAddressProvider.HOST_IP_ADDR_PATH_ENV,
    this.commandBuilder.getIpAddressFilePath()));

for (final int port : this.ports) {
  runOptions.append(String.format("-p %d:%d ", port, port));
}

return new TaskContainerSettings()
    .withRegistry(this.containerRegistryProvider.getContainerRegistry())
    .withImageName(this.containerRegistryProvider.getContainerImageName())
    .withContainerRunOptions(runOptions.toString());
```

Reply:
Fixed.

---
In reply to: [207102310](https://github.com/apache/reef/pull/1478#discussion_r207102310) [](ancestors = 207102310)